pulser/simulation/simulation.py

```diff
@@ -325,55 +535,107 @@ def get_hamiltonian(self, time: float) -> qutip.Qobj:
                              "greater than or equal to 0.")
         return self._hamiltonian(time/1000)  # Creates new Qutip.Qobj
 
+    def _reset_samples(self) -> None:
+        """Sets all samples to their default values.
+
+        Used when running several noisy runs, each one setting new samples.
+        """
+        self.samples = {addr: {basis: {}
+                               for basis in ['ground-rydberg', 'digital']}
+                        for addr in ['Global', 'Local']}
+
+    def _init_dephasing(self) -> None:
+        """Initializes dephasing collapse operators."""
+        if self.basis_name == 'digital' or self.basis_name == 'all':
+            raise NotImplementedError("Cannot include dephasing noise in" +
+                                      " digital- or all-basis.")
+        # Probability of phase (Z) flip
+        prob = self.config.dephasing_prob
+        self._collapse_ops += [
+            np.sqrt(1. - prob) * self.op_matrix['I'],
+            np.sqrt(prob) * (self.op_matrix['sigma_rr']
+                             - self.op_matrix['sigma_gg'])]
+
     # Run Simulation Evolution using Qutip
-    def run(self,
-            initial_state: Optional[Union[np.ndarray, qutip.Qobj]] = None,
-            progress_bar: Optional[bool] = None,
+    def run(self, progress_bar: Optional[bool] = None,
             **options: qutip.solver.Options) -> SimulationResults:
-        """Simulates the sequence using QuTiP's solvers.
+        """Simulate the sequence using QuTiP's solvers.
```

Comment:
I wonder why flake8-docstrings is not complaining about this... In any case, you should use `Simulates` instead